setup/security/authentication/x509/index.md

```diff
@@ -90,14 +90,23 @@ Encoding with any other OID can be done by editing the `openssl.conf`.
 
     >**Note:** If providing multiple groups, as in this example, separate them with a new line (`\n`). The new line `\n` shows as a `%0A` in the certificate.
 
-1. Generate a CSR for a new x509 certificate and the given `openssl.conf`:  
+1. Generate a CSR for a new x509 certificate and the given `openssl.conf`:
     ```
     openssl req -nodes -newkey rsa:2048 -keyout key.out -out client.csr \
         -subj "/C=US/ST=CA/L=Oakland/O=Spinnaker/CN=example@example.com" -config openssl.conf
     ```
+
+1. Create extention config file `extension.conf` to apply roles when signing the server requests.
+    ```
+    [ v3_req ]
+    keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+    1.2.840.10070.8.1 = ASN1:UTF8String:spinnaker-example0\nspinnaker-example1
+    ```
+    The same rule for the roles definition applied to this section, as it's explained in the first step of this section.
+
 1. Use the CA to sign the server's request. (If using an external CA, they do this for you.)
     ```
-    openssl x509 -req -days 365 -in client.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client.crt
+    openssl x509 -req -days 365 -in client.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client.crt -extensions v3_req  -extfile ./extension.conf
```

Comment:
This is actually required - just having it in the base config apparently doesn't work.  Whether it's required as a separate file or with-in the main file and you reference that tends to be a matter of preference.  It's mostly needing the config block in the file referenced via extfiles.

Reply:
for me in this case it's more clear what will be used for this specific command and can be easily checked, but you are right they are duplicating the content with the base file.